project.go

```diff
@@ -41,6 +41,37 @@ func findProjectRoot(from string) (string, error) {
 	}
 }
 
+// checkCfgFilenames validates filename case for the manifest and lock files.
+// This is relevant on case-insensitive systems like Windows.
+func checkCfgFilenames(projectRoot string) error {
+	// ReadActualFilenames is actually costly.  Since this check is not relevant
+	// to case-sensitive filesystems like ext4, try for an early return.
+	if fs.IsCaseSensitiveFilesystem(projectRoot) {
+		return nil
+	}
+
+	actualFilenames, err := fs.ReadActualFilenames(projectRoot, []string{ManifestName, LockName})
+
+	if err != nil {
+		return err
+	}
+
+	// Since this check is done after `findProjectRoot`, we can assume that
+	// manifest file will be present. Even if it is not, error will still be thrown.
+	// But error message will be a tad inaccurate.
+	actualMfName := actualFilenames[ManifestName]
+	if actualMfName != ManifestName {
+		return fmt.Errorf("manifest filename '%s' does not match '%s'", actualMfName, ManifestName)
+	}
+
+	actualLfName := actualFilenames[LockName]
+	if len(actualLfName) > 0 && actualLfName != LockName {
```

Comment:
If a file is not found, the string map returned by `ReadActualFilenames` will have an empty string against the given file name. Since lock file is optional, we should check for equality only if it was found.